File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        List<Integer> lng = a.chars().map(Character::getNumericValue).boxed().collect(Collectors.toList());
        List<Integer> srt = b.chars().map(Character::getNumericValue).boxed().collect(Collectors.toList());
        List<Integer> tm;
        int add = 0;
        int tmp = 0;
        int i = 1;
        if (a.length() < b.length()) {
            tm = lng;
            lng = srt;
            srt = tm;
        }
        while (i <= lng.size()) {
            if (i <= srt.size()) {
                tmp = lng.get(lng.size() - i) + srt.get(srt.size() - i);
                lng.set(lng.size() - i, (tmp + add) % 2);
                add = (tmp + add) / 2;
            } else {
                tmp = lng.get(lng.size() - i);
                lng.set(lng.size() - i, (tmp + add) % 2);
                add = (tmp + add) / 2;
            }
            i ++;
        }
        if (add > 0) {
            lng.add(0, add % 2);
            if (add > 1) {
                lng.add(0, add / 2);
            }
        }
        return lng.stream().map(String::valueOf).collect(Collectors.joining(""));
    }
}

File: 0067-add-binary/0067-add-binary.py
class Solution(object):
    def addBinary(self, a, b):
        mx, mn = list(a[::-1]), list(b[::-1])
        if len(b) >= len(a):
            mx, mn = mn, mx
        for k in range(len(mx)):
            if k < len(mn):
                if k+1 < len(mx):
                    mx[k + 1] = (int(mx[k + 1]) + (int(mx[k]) + int(mn[k])) // 2)
                else:
                    if (int(mx[k]) + int(mn[k])) // 2 > 0:
                        mx.append((int(mx[k]) + int(mn[k])) // 2)
                mx[k] = (int(mx[k]) + int(mn[k])) % 2
            else:
                if int(mx[k]) // 2 > 0:
                    if k+1 < len(mx):
                        mx[k + 1] = int(mx[k + 1]) + int(mx[k]) // 2
                    else:
                        mx.append(int(mx[k]) // 2)
                mx[k] = int(mx[k]) % 2
        if len(mx) == k + 2 and int(mx[k+1]) // 2 > 0:
            mx.append(int(mx[k + 1]) // 2)
            mx[k + 1] = int(mx[k + 1]) % 2
        return "".join(map(str,mx[::-1]))